src/zope/app/apidoc/static.py

```diff
@@ -115,108 +121,156 @@ def isLocalURL(self):
 
     def isApidocLink(self):
         # Make sure that only apidoc links are loaded
-        if self.absoluteURL.startswith(self.rootURL+'++apidoc++/'):
-            return True
-        if self.absoluteURL.startswith(self.rootURL+'@@/'):
+        allowed_prefixes = ((self.rootURL + '++apidoc++/'),
+                            (self.rootURL + '@@/'))
+        if self.absoluteURL.startswith(allowed_prefixes):
             return True
-        return False
-
 
-class OnlineBrowser(mechanize.Browser, object):
 
-    def __init__(self, factory=None, history=None, request_class=None):
-        if factory == None:
-            factory = ApiDocDefaultFactory()
-        mechanize.Browser.__init__(self, factory, history, request_class)
+class OnlineBrowser(zope.testbrowser.browser.Browser):
 
     def setUserAndPassword(self, user, pw):
         """Specify the username and password to use for the retrieval."""
-        hash = base64.encodestring(user+':'+pw).strip()
-        self.addheaders.append(('Authorization', 'Basic '+hash))
+        user_pw = user + ':' + pw
+        if not isinstance(user_pw, bytes):
+            user_pw = user_pw.encode('latin-1')
```

Comment:
latin-1 is the traditional and expected encoding for HTTP headers, isn't it? The newer headers are specified as UTF-8, but the older ones were specified as latin-1.